### DISCUSSION TITLE
Save pattern in Console

### DISCUSSION THREAD

**Discussion by frankfzhao** [twilio-labs/paste#3490] [General] [answered]
Hey Paste team, the Conversations UI team has been working on a experience in Console for customers to configure address and one pattern we would love some input from Paste is the interaction behavior when customers edit the info in a form then save.

Currently I have noticed 2 different save patterns for save a form.

1. When edit an existing record, it stays at the same page, show a toast message stating the change is successful and customers  click a secondary button to go back to the listing page.  And Messaging Service, Phone numbers all follow the same pattern. [See screenshot below]
<img width="2531" alt="Screenshot 2023-09-18 at 12 38 25 PM" src="https://github.com/twilio-labs/paste/assets/82524550/a652aaee-302b-4b87-9f47-280eb4bd0bd5">

2. When edit an existing record, it redirects customers to the listing page, show a toast message stating the change is successful. And Alarm, Intelligent alerts follow this pattern [See screenshot below]
 
<img width="1257" alt="Screenshot 2023-09-18 at 12 41 23 PM" src="https://github.com/twilio-labs/paste/assets/82524550/1457854f-28f2-4d99-8f45-ae5855a2e940">

Any suggestions from paste on this save & redirect pattern? thanks!

**Comment by serifluous** [comment, marked as answer]
You can take a look at our guidelines on Notifications and Feedback: https://paste.twilio.design/patterns/notifications-and-feedback#standard-feedback-indicator-toast and on Creation: https://paste.twilio.design/patterns/create#post-creation

Relevant guidelines:

1. When editing an object and navigating to another page on save, you might not need the Toast. Navigating away from the page may "sufficiently communicate that the information...was saved, without needing further messaging."
2. When creating an object, the guidance is to navigate back to the list page or detail page.

The first pattern you shared isn't one that we've outlined anywhere in Paste, but if it's proved useful to customers, we totally welcome contributions for an "Edit object" pattern.

**Comment by frankfzhao** [comment]
Thanks Sarah! We will dig more into it and circle back.
